### PR TITLE
feat: adding location service mixin

### PR DESCRIPTION
**Pull request by alicejli**
For https://github.com/googleapis/google-cloud-node-core/issues/611

As part of adding support generally for client library mixin method generations (draft AIP is at aip-dev/google.aip.dev#694), this update is to update the generator to add in google.cloud.location.Locations service as a mixin for any api that includes it in their <$service>.yaml file.

Ping me for the internal doc(s) if desired.

### CODE REVIEW

File: templates/typescript_gapic/_locations.njk [hunk 1]
```diff
@@ -0,0 +1,82 @@
+{%- macro locationServiceMethods() -%}
+{#-
+  This part will be added into src/version/client.ts.
+  If the --service-yaml contains "google.iam.v1.IAMPolicy", it means the client requires IAM methods [setIamPolicy, getIamPolicy, testIamPerssion].
+  IamClient is created for the client in the constructor.
+  [setIamPolicy, getIamPolicy, testIamPermission] methods are created which is calling the corresponding methods from IamClient in google-gax.
```
Comment (summer-ji-eng) [inline templates/typescript_gapic/_locations.njk:6]: nit: Should we use the location information here. :)

File: templates/typescript_gapic/_locations.njk [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      LocationProtos.google.cloud.location.ILocation,
+      | LocationProtos.google.cloud.location.IGetLocationRequest
+      | null
+      | undefined,
+      {} | null | undefined
+    >
+  ): Promise<LocationProtos.google.cloud.location.ILocation> {
+    return this.locationsClient.getLocation(request, options, callback);
+  }
+/**
```
Comment (summer-ji-eng) [inline templates/typescript_gapic/_locations.njk:46]: nit: add a line between the methods.

**Review by summer-ji-eng — CHANGES_REQUESTED** [review]
Could you fix the tests and once you done, let me know, I will take another look.
Thanks

File: templates/typescript_gapic/_locations.njk (hunk 1 above)
Comment (alicejli) [reply templates/typescript_gapic/_locations.njk]: great catch! thank you, updated.

File: templates/typescript_gapic/_locations.njk (hunk 2 above)
Comment (alicejli) [reply templates/typescript_gapic/_locations.njk:48]: done.

**Comment by alicejli** [comment]
> Could you fix the tests and once you done, let me know, I will take another look.
> Thanks

Will do! I think they are failing because the gax-nodejs changes aren't merged yet. They should hopefully pass once https://github.com/googleapis/gax-nodejs/pull/1064 is merged.

**Review by summer-ji-eng — APPROVED** [review]

**Pull request merged**